src/disambiguate-series-statements.js

```diff
@@ -0,0 +1,218 @@
+
+    const deletableFieldsAsStrings = deletableFields.map(f => fieldToString(f));
+
+    deletableFields.forEach(nf => nvdebug(`TODO: remove field '${fieldToString(nf)}'`, debug));
+
+    deletableFields.forEach(f => record.removeField(f));
+
+    return {message: [], fix: deletableFieldsAsStrings, valid: true};
+  }
+
+  async function validate(record) {
+    const deletableFields = await getDeletableFields(record);
+
+    if (deletableFields.length === 0) {
+      return {'message': [], 'valid': true};
+    }
+
+    const deletableFieldsAsStrings = deletableFields.map(f => `REMOVE '${fieldToString(f)}'`);
+
+    return {'message': deletableFieldsAsStrings, 'valid': false};
+  }
+
+  async function getDeletableFields(record) {
+    const recordType = getRecordType(record);
+    //console.info(`getDeletableFields(record), type: ${recordType}`); // eslint-disable-line no-console
+    if (recordType === UNDEFINED) {
+      return [];
+    }
+
+    const seriesStatements = record.get('490').filter(f => fieldHasSubfield(f, 'x'));
+
+    //console.info(`N STATEMENTS=${seriesStatements.length}`); // eslint-disable-line no-console
+    if (seriesStatements.length < 2) {
+      return [];
+    }
+
+    const deletableFields = await getDeletableSeriesStatementFields(recordType, seriesStatements);
+    // Delete something if and only if we keep something as well:
+    if (deletableFields.length < seriesStatements.length) {
+      return deletableFields;
+    }
+
+    return [];
+  }
+
+  async function getDeletableSeriesStatementFields(recordType, seriesStatementFields, deletableFields = []) {
+    //console.info(`getDeletableSeriesStatementFields(), N CANDS=${seriesStatementFields.length}`); // eslint-disable-line no-console
+
+    if (seriesStatementFields.length === 0) {
+      //console.info(` DONE`); // eslint-disable-line no-console
+      return deletableFields;
+    }
+
+    const [currStatementField, ...rest] = seriesStatementFields;
+
+    const removeMe = await isRemovableField(currStatementField);
+    //console.info(` ${removeMe ? 'REMOVE' : 'KEEP'}: ${fieldToString(currStatementField)}`); // eslint-disable-line no-console
+
+    const deletableFields2 = removeMe ? [...deletableFields, currStatementField] : deletableFields;
+
+    const result = await getDeletableSeriesStatementFields(recordType, rest, deletableFields2);
+    return result;
+
+    async function isRemovableField(seriesStatementField) {
+      //console.info(` isRemovableField() in...`); // eslint-disable-line no-console
+      const issn = getIssn(seriesStatementField);
+      if (!issn) {
+        return false;
+      }
+      //console.info(` got ISSN ${issn}`); // eslint-disable-line no-console
+      const issnRecords = await issnToRecords(issn);
+      //console.info(` ISSN returned ${issnRecords.length} record(s)`); // eslint-disable-line no-console
+      // TEE: konvertoi tietueet objekteiksi!
+      if (issnRecords.some(r => !isMismatchingRecord(r))) {
```

Comment:
Readability proposal: isn't !isMismatchingRecord same as isMatchingRecord?

Reply:
Also you might use intermediate variables and return the conditional directly, e.g.,
```js
const matchingRecord = issnRecords.find(r => isMatchingRecord(r))
return matchingRecord !== undefined
```

Reply:
"Readability proposal: isn't !isMismatchingRecord same as isMatchingRecord?"

This was deliberate compromise: as UNDEFINED (const for 0) record type does not mean it's a matching record. UNDEFINED does not trigger match nor mismatch... so isMatchingRecord is not same as !isMismatchingRecord... Readability is admittedly iffy, and there might be better ways to do this.